ui/app/components/auth/fields.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+
+/**
+ * @module Auth::Fields
+ *
+ * @example
+ * <Auth::Fields @loginFields={{array "username" "password"}} />
+ *
+ * @param {array} loginFields - array of strings to render as input fields
+ */
+
+export default class AuthFields extends Component {
```

Comment:
Originally this was a template only component, but then the autocomplete logic got fancier so opted to make a backing class so the template logic was easier to parse.